palette/src/luma/relative_contrast.rs

```diff
@@ -0,0 +1,34 @@
+pub mod wcag {
+    use Component;
+
+    /// A trait for calculating relative contrast between two colors.
+    ///
+    /// W3C's Web Content Accessibility Guidelines (WCAG) 2.1 suggest a method
+    /// to calculate accessible contrast ratios of text and background colors
+    /// for those with low vision or color deficiencies. The possible range of
+    /// ratios is from 1:1 to 21:1. There is a Success Criterion for Contrast
+    /// (Minimum) and a Success Criterion for Contrast (Enhanced), SC 1.4.3 and
+    /// SC 1.4.6 respectively. The relative contrast is calculated by `(L1 +
+    /// 0.05) / (L2 + 0.05)`, where `L1` is the luminance of the brighter color
+    /// and `L2` is the luminance of the darker color both in sRGB linear space.
+    /// Higher contrast ratio is generally desireable.
+    pub trait RelativeContrast {
+        /// The return type for the contrast ratio.
+        type Scalar: Component;
+
+        /// Calculate contrast ratio between two colors.
+        fn get_contrast_ratio(&self, other: &Self) -> Self::Scalar;
+        /// Verify the contrast between two colors satisfies SC 1.4.3. Contrast
+        /// is at least 4.5:1 (Level AA).
+        fn is_min_contrast(&self, other: &Self) -> bool;
```

Comment:
I think this and the ones below can have default implementations. They are going to be the same, regardless of the color representation, right?